helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroRecordUtil.java

```diff
@@ -230,6 +231,69 @@ public static GenericRecord specificRecordToGenericRecord(
     return outputRecord;
   }
 
+  public static <T extends IndexedRecord> T setStringField(T record, String fieldName, CharSequence value) {
+    Schema schema = record.getSchema();
+    Schema.Field field = schema.getField(fieldName);
+    if (field == null) {
+      throw new IllegalArgumentException("schema " + schema.getFullName() + " has no such field " + fieldName);
+    }
+    if (!(record instanceof SpecificRecord)) {
+      //generic record
+      //TODO - make this honor schema logical types or runtime avro version?
+      record.put(field.pos(), value);
+      return record;
+    }
+    Class<? extends IndexedRecord> recordClass = record.getClass();
+    //look for public field 1st
+    Field[] classFields = recordClass.getFields();
+    for (Field classField : classFields) {
+      //TODO - take into account avro name mangling?
```

Comment:
+1. method used in Codegen can be made common.

Reply:
will be done as part of a follow-up